lib/specifications/types/Application.js

```diff
@@ -124,15 +124,15 @@ class Application extends ComponentProject {
 		try {
 			return await this._getNamespaceFromManifestJson();
 		} catch (manifestJsonError) {
-			if (manifestJsonError.code !== "ENOENT") {
+			if (manifestJsonError.cause?.code !== "ENOENT") {
```

Comment:
In general, I don't like if the caller needs to know the potential structure of an exception chain. However, as this is inside the same module, it's  acceptable. 

Alternatively, the catch clause starting in line 224 could check for the ENOENT and throw the `err` "as is", without wrapping it. Both error messages anyhow have the same content (no added value).

For the caller this then would result in a direct check on the caught manifestJsonError.

Reply:
Good idea, somehow I missed that 👍